azureLogIngestion/index.ts

```diff
@@ -1,21 +1,24 @@
 import { AzureFunction, Context } from "@azure/functions"
-
 import Adapter from "./adapter"
 import OpenTelemetryAdapter from "./opentelemetry"
+import * as _ from "lodash"
 
 const apiKey = process.env["NEW_RELIC_INSERT_KEY"]
 const debug = process.env["DEBUG"] || false
 const otel = process.env["OTEL"] || false
 
 const adapter = new Adapter(apiKey)
-const otelAdapter = new OpenTelemetryAdapter(apiKey)
+let otelAdapter: OpenTelemetryAdapter
 
 const eventHubTrigger: AzureFunction = async function (context: Context, eventHubMessages: any): Promise<void> {
     if (debug) {
         context.log(`Eventhub trigger function called for message eventHubMessages ${eventHubMessages}`)
         context.log(`eventHubMessages type: ${typeof eventHubMessages}`)
     }
     if (otel) {
+        const serviceName = _.get(eventHubMessages, "records.0.resourceId", null)
```

Comment:
We have `./utils/resource` to parse this. We probably want `resourceName` only here.